optuna/_convert_positional_args.py

```diff
@@ -6,11 +6,16 @@
 from typing import TypeVar
 import warnings
 
-from typing_extensions import ParamSpec
+
+try:
+    from typing_extensions import ParamSpec
+
+    _P = ParamSpec("_P")
+except ImportError:
+    _P = Any
```

Comment:
When I execute `import optuna` without `typing-extensions`, I got the following error:
```
  File "optuna/optuna/_experimental.py", line 48, in <module>
    ) -> Callable[[Callable[FP, FT]], Callable[FP, FT]]:
  File "/usr/lib/python3.8/typing.py", line 813, in __getitem__
    raise TypeError(f"Callable[args, result]: args must be a list."
TypeError: Callable[args, result]: args must be a list. Got typing.Any
```

I guess `Callable[args, result]` requirs a list as args and `Any` is not acceptable here.

```suggestion
    _P = [Any]
```

Reply:
Thank you for your review :bow:
I pushed 6f3b807 to fix the problem.